### ISSUE TITLE
[interface-name-prefix] Add exclusions list

### ISSUE THREAD

**Issue by Sasha-Sorokin** [typescript-eslint/typescript-eslint#1216]
**Idea**

Sometimes you want to define global variables in `window` object, to do this you have to declare `global` and extend interface `Window` in it. Unfortunately, ESlint gets triggered in that case if you have `interface-name-prefix` set to always prefix with `I`. The idea is to add list of excluded names that are allowed to be (not) prefixed with `I`.

I also thought of idea if it could be possible to check if interface is being extended or declared in `declare` block, but that's just too much work and I don't think that will be much useful.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/interface-name-prefix": ["error", {
      "prefixWithI": "always",
      "exclude": ["Window"]
    }]
  }
}
```

```TS
declare global {
  interface Window {
    i18n(key: string): string;
  }
}

// ...
```

**Expected Result**: Window is being excluded from the check and not given any warning.

**Actual Result**: Warning produced
```
/tmp/interface-name-example/example.ts
  2:12  error  Interface name must be prefixed with "I"  @typescript-eslint/interface-name-prefix

✖ 1 problem (1 error, 0 warnings)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.7.0` |
| `@typescript-eslint/parser`        | `2.7.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `12.13.0` |
| `npm`                              | `6.10.1` |

**Comment by bradzacher** [comment]
Ignore options are something we avoid.
They lack context and are thus too far reaching, meaning your code can easily become inconsistent.

In this sort of edge case, it's pretty straightforward to use a one-off eslint disable comment.

**Comment by Sasha-Sorokin** [comment]
Okay. Strangely though, what I noticed, there is no quick fix available that will put “disable” comment. Is it fine for `typescript-eslint`?

**Comment by bradzacher** [comment]
If you're using the latest vscode with the latest extension version, it should have it:
![image](https://user-images.githubusercontent.com/7462525/69012161-a7188d00-0927-11ea-8042-2548bf10eb40.png)

**Comment by Sasha-Sorokin** [comment]
I guess something wrong on my side then, will try find what's wrong with my installation. Thanks for checking!
